packages/create-app/index.js

```diff
@@ -39,15 +39,42 @@ async function init() {
   let targetDir = argv._[0]
   if (!targetDir) {
     /**
-     * @type {{ name: string }}
+     * @type {{ projectName: string }}
      */
-    const { name } = await prompt({
+    const { projectName } = await prompt({
       type: 'input',
-      name: 'name',
+      name: 'projectName',
       message: `Project name:`,
       initial: 'vite-project'
     })
-    targetDir = name
+    targetDir = projectName
+  }
+  let packageName
+  const packageNameRegExp = /^(?:@[a-z0-9-*~][a-z0-9-*._~]*\/)?[a-z0-9-~][a-z0-9-._~]*$/
+  if (packageNameRegExp.test(targetDir)) {
+    packageName = targetDir
+  } else {
+    const suggestPkgName = targetDir
+      .trim()
+      .toLowerCase()
+      .replace(/\s+/g, '-')
+      .replace(/^[._]/, '')
+      .replace(/[^a-z0-9-~]+/g, '-')
+
+    /**
+     * @type {{ inputPackageName: string }}
+     */
+    const { inputPackageName } = await prompt({
+      type: 'input',
+      name: 'inputPackageName',
+      message: `Name for package.json:`,
```

Comment:
I think we should use `Package name:` for the input message to be consistent with `Project Name:`

Reply:
I'm worried that it would be not descriptive enough, especially for new users scaffolding their first project.
Part of this prompt's job is to communicate a potential problem. `Package name:` feels a bit ambiguous as to it can be understood as "import package name of a template". I agree that `Name for package.json:` feels clunky though.

Reply:
I think that `Package name` is enough in this context, but maybe we could use `package.json name:` if you think that would help. Another option is to leave it as `Package name` and modify the error message as `Invalid package.json name`, so if there is an issue, it is easy for the user to search about this

Reply:
I like the third option modifiying the error message

Reply:
Done